src/frontend/komponenter/Fagsak/Personlinje/Behandlingsmeny/LeggTilBarnPåBehandling/LeggTilBarnPåBehandling.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from 'react';
+
+import styled from 'styled-components';
+
+import Alertstripe from 'nav-frontend-alertstriper';
+import Hjelpetekst from 'nav-frontend-hjelpetekst';
+import KnappBase, { Flatknapp, Knapp } from 'nav-frontend-knapper';
+import { SkjemaGruppe } from 'nav-frontend-skjema';
+import { Undertittel } from 'nav-frontend-typografi';
+
+import { FamilieInput } from '@navikt/familie-form-elements';
+import { useHttp } from '@navikt/familie-http';
+import { ok, useFelt, useSkjema } from '@navikt/familie-skjema';
+import {
+    byggFeiletRessurs,
+    byggHenterRessurs,
+    byggTomRessurs,
+    Ressurs,
+    RessursStatus,
+} from '@navikt/familie-typer';
+
+import { useFagsakRessurser } from '../../../../../context/FagsakContext';
+import { IBehandling } from '../../../../../typer/behandling';
+import { IFagsak } from '../../../../../typer/fagsak';
+import { adressebeskyttelsestyper, IPersonInfo, IRestTilgang } from '../../../../../typer/person';
+import { identValidator } from '../../../../../utils/validators';
+import UIModalWrapper from '../../../../Felleskomponenter/Modal/UIModalWrapper';
+
+const LeggTilBarnLegend = styled.div`
+    margin-top: 1rem;
+    display: flex;
+`;
+
+interface IProps {
+    onListElementClick: () => void;
+    behandling: IBehandling;
+}
+
+const LeggTiLBarnPåBehandling: React.FC<IProps> = ({ onListElementClick, behandling }) => {
+    const { request } = useHttp();
+    const { settFagsak } = useFagsakRessurser();
+
+    const [visModal, settVisModal] = useState<boolean>(false);
+
+    const { skjema, settSubmitRessurs, kanSendeSkjema, nullstillSkjema } = useSkjema<
+        {
+            ident: string;
+        },
+        IPersonInfo
+    >({
+        felter: {
+            ident: useFelt<string>({
+                verdi: '',
+                valideringsfunksjon:
+                    process.env.NODE_ENV === 'development' ? felt => ok(felt) : identValidator,
+            }),
+        },
+        skjemanavn: 'Legg til barn',
+    });
+
+    const onAvbryt = () => {
+        settVisModal(false);
+        settSubmitRessurs(byggTomRessurs());
+    };
+
+    const leggTilOnClick = () => {
```

Comment:
Kunne vurdert å lage en useLeggTilBarn hook for dette kanskje?

Reply:
Mhm, tenkte også det kunne vært nice, men de er såpass ulike at man måtte sendt inn en onSuccess-funksjon som skjer etter tilgangskallet. Denne vil igjen ha en ytterligere onSuccess i LeggTilBarn, så jeg landa egentlig på det ville vært mer for generaliseringens skyld enn at det egentlig ble noe særlig lesbart og enkelt å jobbe med